src/app/index.ts

```diff
@@ -213,6 +209,10 @@ export class Auth {
         $("#hideSignUp").hide();
         $("#loggedInUserName").removeClass('hidden');
         $("#logoutAction").removeClass('hidden');
+
+        // Start redirect only after response from user is complete.
+        window.location.href = `/_gettingstarted`;
```

Comment:
although this works, doesn't this violate [SRP](https://en.wikipedia.org/wiki/Single_responsibility_principle) i.e. updateUserMenu updates userMenu and redirects the user? I was thinking if updateUserMenu could return a promise and resolve it after the menu is updated. so that the caller could write 
```typescript
updateUserMenu().then(redirect)
```

Reply:
Yes , agreed - Ideally it should be happening by passing a 'callback' function which should be called on completion of the request.